ui/src/components/FileUpload/index.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import * as DropZone from 'react-dropzone';
+import * as OutlineIcons from '@heroicons/react/outline';
+
+import * as Interfaces from '@interfaces';
+import * as Components from '@components';
+import * as Helpers from '@helpers';
+
+type Props = {
+    positiveActionCallback: (files: Interfaces.ImagePreview[]) => void;
+    negativeActionCallback: () => void;
+    loading: boolean;
+};
+
+const FileUpload: React.FC<Props> = (props): React.ReactElement => {
+    const [previewBase64, setPreviewBase64] = React.useState<Interfaces.ImagePreview[]>([]);
+
+    const onDrop = React.useCallback(async (acceptedFiles: Omit<FileList, 'item'>) => {
+        if (acceptedFiles.length) {
+            for (const file of Array.from(acceptedFiles)) {
+                const base64 = await Helpers.getBase64FromFile(file);
+                setPreviewBase64((prevState) => [...prevState, { name: file.name, base64 }]);
+            }
+        }
+    }, []);
+
+    const { getRootProps, getInputProps, isDragActive } = DropZone.useDropzone({ onDrop });
+
+    return (
+        <section key="file-upload" className="relative">
```

Comment:
Can we remove `key` from the section, or is it used?

Reply:
Applied in latest commit